spec/components/promotion_spec.rb

```diff
@@ -102,6 +102,35 @@
       end
     end
 
+    context "may send tl2 promotion messages" do
+      fab!(:user) { Fabricate(:user, trust_level: TrustLevel[1], created_at: (SiteSetting.tl2_requires_time_spent_mins * 60).minutes.ago) }
+
+      before do
+        stat = user.user_stat
+        stat.topics_entered = SiteSetting.tl2_requires_topics_entered
+        stat.posts_read_count = SiteSetting.tl2_requires_read_posts
+        stat.time_read = SiteSetting.tl2_requires_time_spent_mins * 60
+        stat.days_visited = SiteSetting.tl2_requires_days_visited
+        stat.likes_received = SiteSetting.tl2_requires_likes_received
+        stat.likes_given = SiteSetting.tl2_requires_likes_given
+        stat.stubs(:calc_topic_reply_count!).returns(SiteSetting.tl2_requires_topic_reply_count)
```

Comment:
Usually we prefer to not have to stub. Is it possible to set the site setting to 0?